Automattic/pocket-casts-android
# Add logic routing to appropriate UI when giving a rating

## Description
This PR checks whether a user is signed in, connected to the internet, and can rate a podcast (logic for being able to rate a podcast will probably be modified in the future), and based on that shows the appropriate placeholder screen.

In addition, this adds a Loading screen to display until we know whether the user can rate the podcast. This loading screen is the only UI in this PR that is not placeholder UI.

## Testing Instructions

1. Make sure that the ratings beta flag is turned on (it's on by default on debug builds)
3. Go to a podcast with a rating
4. Tap on the star rating
5. See that you are shown a screen for rating the podcast that mathes the designs (sQk9KlwngN1g8iJbQGAMva-fi-23_2052). Note that the stars portion of this screen is a placeholder that will be replaced in a follow-up PR.

## Checklist
